api/hypershift/v1beta1/hosted_controlplane.go

```diff
@@ -184,6 +184,15 @@ type HostedControlPlaneSpec struct {
 	//
 	// +optional
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// labels when specified, define what custome labels are added to the hcp pods.
+	// Changing this day 2 will cause a rollout of all hcp pods.
+	// Valid values are those in https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+	//
+	// +kubebuilder:validation:XValidation:rule="self.all(key, (key.split('/', 2).size() == 2 ? key.split('/', 2).first().size <= 253 && key.split('/', 2).first().matches('^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$') : true) && key.split('/', 2).last().size <= 63 && key.split('/', 2).last().matches('^([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$'))", message="label key must have two segments: an optional prefix and name, separated by a slash (/). The name segment is required and must be 63 characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between. The prefix is optional. If specified, the prefix must be a DNS subdomain: a series of DNS labels separated by dots (.), not longer than 253 characters in total, followed by a slash (/)"
+	// +kubebuilder:validation:XValidation:rule="self.all(key, size(self[key]) <= 63 && self[key].matches('^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$'))", message="label value must be 63 characters or less (can be empty), consist of alphanumeric characters, dashes (-), underscores (_) or dots (.), and begin and end with an alphanumeric character"
+	// +optional
```

Comment:
please include maxProperties

Reply:
e.g.
// +kubebuilder:validation:MaxProperties=10

Also what happens if I introduce the same label key more than once?
Can we please also check uniqueness?

Reply:
> Also what happens if I introduce the same label key more than once?
> Can we please also check uniqueness?

this is a map, keys are implicitly validated to be unique, no?

Reply:
updated the godoc

Reply:
So if duplicate entries are passed, latest one take prio? or something else?
Also what happens if the input passed is not valid? does this fail in a condition?

Reply:
done